ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSfilemanip.cpp

```diff
@@ -107,92 +114,454 @@ void ini_section_delete(std::string section)
 
 /* OS Specific; should be moved */
 
-int file_exists(std::string fname) {
-    DWORD attributes = GetFileAttributes(fname.c_str());
-    if(attributes == 0xFFFFFFFF) {
-        return 0;
-    } else {
-        return 1;
-    }
+int file_exists(std::string fname)
+{
+    std::string str(fname);
+    std::replace(str.begin(),str.end(),'/','\\');
+
+    struct stat sb;
+    return (stat((char *)str.c_str(),&sb) == 0 &&
+        S_ISREG(sb.st_mode));
 }
 
-int file_delete(std::string fname) {
-    DWORD result = DeleteFileA(fname.c_str());
-
-    switch(result) {
-        case 0:
-            return 0;
-            break;
-        case ERROR_FILE_NOT_FOUND:
-            return 0;
-            break;
-        case ERROR_ACCESS_DENIED:
-            return 0;
-            break;
-        default:
-            return 1;
-            break;
+int file_delete(std::string fname)
+{
+    std::string str(fname);
+    std::replace(str.begin(),str.end(),'/','\\');
+
+    struct stat sb;
+    if (stat((char *)str.c_str(),&sb) == 0 &&
+        S_ISREG(sb.st_mode))
+    {
+        return (remove((char *)str.c_str()) == 0);
```

Comment:
The `remove` function is also POSIX... which Windows is not. The way the function was was fine, though, again, you may want to use `DeleteFileW` and `MultiByteToWideChar` to allow UTF-8.

I'd also remove the redundant `break;` commands in the old function....